pinecone/client.go

```diff
@@ -80,14 +207,116 @@ func NewClientBase(in NewClientBaseParams) (*Client, error) {
 	return &c, nil
 }
 
-func (c *Client) Index(host string) (*IndexConnection, error) {
+// Index creates an IndexConnection to the specified host.
+//
+// Parameters:
+//   - host: The host URL of your Pinecone index.
+//
+// Returns a pointer to an IndexConnection instance on success. In case of failure, it returns nil and an error.
+//
+// Example:
+//  ctx := context.Background()
+//
+//  clientParams := pinecone.NewClientParams{
+//    ApiKey:    getEnvVars("PINECONE_API_KEY"),
+//    SourceTag: "your_source_identifier", // optional
+//   }
+//
+//  pc, err := pinecone.NewClient(clientParams)
+//  if err != nil {
+//    log.Fatalf("Failed to create Client: %v", err)
+//   }
+//
+//  idxs, err := pc.ListIndexes(ctx)
+//	if err != nil {
+//	  fmt.Println("Error:", err)
+//	 }
+//
+//  idx, err := pc.Index(idxs[0].Host) // You can now use idx to interact with your index.
+//  defer idx.Close()
+//
+//  if err != nil {
+//    fmt.Println("Error:", err)
+//   }
+ func (c *Client) Index(host string) (*IndexConnection, error) {
 	return c.IndexWithAdditionalMetadata(host, "", nil)
 }
 
+// IndexWithNamespace creates an IndexConnection to the specified host within the specified namespace.
+//
+// Parameters:
+//   - host: The host URL of your Pinecone index.
+//   - namespace: The target namespace.
+//
+// Returns a pointer to an IndexConnection instance on success. In case of failure, it returns nil and an error.
+//
+// Example:
+//  ctx := context.Background()
+//
+//  clientParams := pinecone.NewClientParams{
+//    ApiKey:    getEnvVars("PINECONE_API_KEY"),
+//    SourceTag: "your_source_identifier", // optional
+//   }
+//
+//  pc, err := pinecone.NewClient(clientParams)
+//  if err != nil {
+//    log.Fatalf("Failed to create Client: %v", err)
+//   }
+//
+//  idxs, err := pc.ListIndexes(ctx)
+//	if err != nil {
+//	  fmt.Println("Error:", err)
+//	 }
+//
+//  idx, err := pc.IndexWithNamespace(idxs[0].Host, <"sample-namespace">) // You can now use idx to interact with your index.
+//  defer idx.Close()
+//
+//  if err != nil {
+//    fmt.Println("Error:", err)
+//   }
 func (c *Client) IndexWithNamespace(host string, namespace string) (*IndexConnection, error) {
 	return c.IndexWithAdditionalMetadata(host, namespace, nil)
 }
 
+// IndexWithAdditionalMetadata creates an IndexConnection to the specified host within the specified namespace,
+// with the addition of custom metadata fields.
```

Comment:
`additionalMetadata` can be thought of analogous to `Headers` in the `NewClient` function. They're metadata that's applied to each RPC request to the server: https://grpc.io/docs/guides/metadata/. I think we could be explicit about that here, and maybe even link to some of the supporting gRPC documentation.